examples/cpp/pyperf/PyPerfNativeStackTrace.cc

```diff
@@ -200,5 +226,36 @@ bool NativeStackTrace::error_occured() const {
   return error_occurred;
 }
 
+void NativeStackTrace::prune_dead_pid(uint32_t dead_pid) {
+  auto it = procSymbolsCache.find(dead_pid);
+  if (it != procSymbolsCache.end()) {
+    procSymbolsCache.erase(it);
+  }
+}
+
+void NativeStackTrace::enable_dso_reporting() {
+  NativeStackTrace::insert_dso_name = true;
+}
+
+ProcSyms* NativeStackTrace::get_proc_symbols(uint32_t pid) {
+  static struct bcc_symbol_option symbol_options = {
+    .use_debug_file = 1,
+    .check_debug_file_crc = 1,
+    .lazy_symbolize = 1,
+    .use_symbol_type = BCC_SYM_ALL_TYPES
+  };
+  auto current_time_point = std::chrono::steady_clock::now();
+  double timestamp_s = std::chrono::duration_cast<std::chrono::duration<double>>(current_time_point - startup_time_point).count();
```

Comment:
You don't need to subtract `startup_time_point` - I don't think that the latter is needed at all, just check `now() - procSymbolsCache[pid].timestamp_s > ProcSymbolsCacheTTL_S` and that should be enough, no?

Reply:
Good idea. 
Direct cast is not available, but a simple workaround can be built with `time_point::time_since_epoch()`.